src/routes.js

```diff
@@ -805,8 +827,16 @@ const routesByName = new Map();
       : false)
   );
 
-
-
+  [
+    'NewSubmission',
+    'NewDraftSubmission',
+    'EditSubmission',
+    'FormPreview',
+    'DraftFormPreview'
+  ].forEach(redirectTo => {
+    const preserve = (_, from) => from.name === 'EnketoRedirector' && [form];
+    routesByName.get(redirectTo).meta.preserveData.push(preserve);
+  });
```

Comment:
can't use `preserveDataBetweenRoutes` because it add preserveData for all the routes provided as first argument. We want to preserveData one way not between all.